service/javadsl/server/src/main/java/com/lightbend/lagom/javadsl/server/ServiceGuiceSupport.java

```diff
@@ -3,23 +3,49 @@
  */
 package com.lightbend.lagom.javadsl.server;
 
-import com.lightbend.lagom.internal.javadsl.*;
+import com.google.inject.Binder;
+import com.lightbend.lagom.internal.javadsl.BinderAccessor;
+import com.lightbend.lagom.internal.javadsl.server.JavadslServicesRouter;
 import com.lightbend.lagom.internal.javadsl.server.ResolvedServices;
 import com.lightbend.lagom.internal.javadsl.server.ResolvedServicesProvider;
 import com.lightbend.lagom.internal.javadsl.server.ServiceInfoProvider;
-import com.lightbend.lagom.internal.javadsl.server.JavadslServicesRouter;
 import com.lightbend.lagom.internal.server.status.MetricsServiceImpl;
-
-import com.lightbend.lagom.javadsl.server.status.MetricsService;
-import com.google.inject.Binder;
 import com.lightbend.lagom.javadsl.api.ServiceInfo;
 import com.lightbend.lagom.javadsl.client.ServiceClientGuiceSupport;
+import com.lightbend.lagom.javadsl.server.status.MetricsService;
+
+import java.util.Arrays;
+import java.util.Optional;
 
 public interface ServiceGuiceSupport extends ServiceClientGuiceSupport {
 
+    /**
+     * @deprecated use {@link ServiceGuiceSupport#bindServices(String, ServiceBinding[])} instead.
+     */
     default void bindServices(ServiceBinding<?>... serviceBindings) {
+        bindServices(Optional.empty(), serviceBindings);
+    }
+
+    /**
+     * binds the provided services and builds the {@link ServiceInfo} for this service.
+     *
+     * @param serviceName
+     * @param serviceBindings
+     */
+    default void bindServices(String serviceName, ServiceBinding<?>... serviceBindings) {
+        bindServices(Optional.of(serviceName), serviceBindings);
+    }
+
+    /**
+     * binds the provided services and builds the {@link ServiceInfo} for this service. If <code>serviceName</code>
+     * is an {@link Optional#empty()}, the {@link ServiceInfo#serviceName} will be set to the name of the first
+     * serviceBinding. Passing an {@link Optional#empty()} as <code>serviceNam</code> is discouraged.
+     *
+     * @deprecated this method is an implementation detail and will likely bbe removed in future versions, use {@link ServiceGuiceSupport#bindServices(String, ServiceBinding[])} instead.
+     */
+    default void bindServices(Optional<String> serviceName, ServiceBinding<?>... serviceBindings) {
         Binder binder = BinderAccessor.binder(this);
-        for (ServiceBinding binding: serviceBindings) {
+        for (ServiceBinding binding : serviceBindings) {
```

Comment:
When adding a String on `bindServices ` I extracted the code and created `bindServices(Optional<String> serviceName, ServiceBinding<?>... serviceBindings)` which (unfortunately) is public. This should be removed when we remove `bindService(ServiceBinding...)`